packages/x-date-pickers/src/locales/utils/pickersLocaleTextApi.ts

```diff
@@ -26,14 +41,23 @@ export interface PickersLocaleText<TDate> {
   hoursClockNumberText: (hours: string) => string;
   minutesClockNumberText: (minutes: string) => string;
   secondsClockNumberText: (seconds: string) => string;
+
+  // Open picker labels
   openDatePickerDialogue: (date: TDate | null, utils: MuiPickersAdapter<TDate>) => string;
   openTimePickerDialogue: (date: TDate | null, utils: MuiPickersAdapter<TDate>) => string;
+
+  // Table labels
   timeTableLabel: string;
   dateTableLabel: string;
-  datePickerDefaultToolbarTitle: string;
-  dateTimePickerDefaultToolbarTitle: string;
-  timePickerDefaultToolbarTitle: string;
-  dateRangePickerDefaultToolbarTitle: string;
+
+  // Field section placeholders
+  fieldYearPlaceholder: (params: { digitAmount: number }) => string;
```

Comment:
I kept a method format for all sections in case we need to pass params in the future (to avoid breaking changes)

Right now I have 3 sections where it feel a bit weird if I always return the placeholder

- The year: if the year to write is `22` I put `YY`, if it is `2022` I put `YYYY`

- The month: if the month to write is `August` I put `MMMM`, if it is `08` or `8` I put `MM`. One could say that for `Aug` we should write `MMM`, but that would be harder to determine and probably less reliable in other languages.

Reply:
I like the idea. 

I assume those methods could receive the `contentType` and the `token`. Such that developers can easily customize placeholders at the level of the app instead of the component level.

Reply:
Yes, some params probably make no sense to some sections, so I did not add them everywhere.
But that's something we can add whenever a user asks for it.

For the `token`, nothing prevents us from passing it. I feel like most of the time we don't want to rely on it because because it depends on the library.
So it should only be used in the app, not in our code (including in our locales).
Maybe we could start without it to give users good habit and add it if they ask for it.

For the `contentType`, the only one that can have both is the month and I'm already passing it.